packages/blocks/src/blocks/core-template-part.tsx

```diff
@@ -20,13 +20,17 @@ const CoreTemplatePart: CoreTemplatePartType = ( {
 	children,
 	attributes,
 }: CoreTemplatePartProps ) => {
-	const { templatePartTagName } = attributes || {};
-	/**
-	 * @todo add support for area_tag.
-	 * @see https://github.com/WordPress/gutenberg/blob/4775e7052b9e2ed7df46429e6e738de3faf2fb18/packages/block-library/src/template-part/index.php#L165
-	 */
-	const TagName = ( templatePartTagName ||
-		'div' ) as keyof JSX.IntrinsicElements;
+	const { templatePartTagName, area } = attributes || {};
+
+	let htmlTag = 'div';
+
+	if ( templatePartTagName ) {
+		htmlTag = templatePartTagName;
+	} else if ( typeof area === 'string' && area.trim() !== '' ) {
```

Comment:
Any reason not to do the same checks for `templatePartTagName`?

Reply:
Better q: any reason to use a default value and a double conditional vs `htmlTag = templatePartTagName || area || 'div'` ?

(But either way we don't need to worry about trimming. )